cypress/support/service/administration/fixture/newsletter-recipient.fixture.js

```diff
@@ -0,0 +1,42 @@
+const AdminFixtureService = require('../fixture.service.js');
+
+class NewsletterRecipient extends AdminFixtureService {
+    setNewsletterRecipientFixture(recipient) {
+
+        cy.getSalesChannelId().then((accessKey) => {
+            const headers = {
+                "Content-Type": "application/json",
+                "SW-Access-Key": accessKey
+            };
+
+            function getSalutation() {
+                const url = '/sales-channel-api/v3/salutation';
+                return fetch(url, { method: 'GET', headers })
+                    .then((resp) => resp.json())
+                    .then((json) => json.data[0]);
+            }
+
+            function subscribeRecipient(recipient) {
+                const url = '/sales-channel-api/v3/newsletter/subscribe';
+                const body = JSON.stringify(recipient);
+                return fetch(url, { method: 'POST', headers, body })
```

Comment:
Don't use `fetch` here. `fetch` is not available in Node.js and will be polyfilled with `node-fetch` - see: https://www.npmjs.com/package/node-fetch. Personally I don't see a need to use `fetch` at all. `cy.request()`is what we're using throughout the different kinds of fixture services which worked great for us in the past.

Please refer to the documentation if you have questions how to integrate it: https://docs.cypress.io/api/commands/request.html#Arguments

Additionally you're extending from `AdminFixtureService` which extends from `ApiService` itself. Here we're creating a HTTP client using Axios which can be used as well. Refer to the `ApiService` class regarding the implementation: https://github.com/shopware/e2e-testsuite-platform/blob/master/cypress/support/service/api.service.js

Reply:
Agreed, just one little remark here: `cy.request` is not available in the services. Therefore I suggest Promises in our axios context.